src/pages/members.tsx

```diff
@@ -14,17 +14,65 @@ import {
 import theme from '../theme';
 import constants from '../util/strings/constants';
 import { AppContext } from './_app';
-import FooterPatternIcon from '../components/icons/FooterPatternIcon';
 import FooterPattern from '../components/FooterPattern';
+import { useRouter } from 'next/router';
+import { getMembers } from '../services/APIService';
+import { logError } from '../util/sentry';
 
 function Members() {
     const {
+        authToken,
+        setAuthToken,
+        setMembers,
+        setFamilyManagerEmail,
+        setTotalStorage,
+        setMessage,
+        setMessageDialogView,
         isLargerDisplay,
         familyManagerEmail,
         members,
         inviteDialogView,
         setInviteDialogView,
     } = useContext(AppContext);
+
+    const router = useRouter();
+
+    useEffect(() => {
+        if (authToken) {
+            syncMembers();
+        }
+    }, [authToken]);
+
+    const syncMembers = async () => {
+        try {
+            const res = await getMembers(authToken);
+            if (res.success) {
+                setMembers(res.data.members);
+                setTotalStorage(res.data.storage);
+                for (const member of res.data.members) {
+                    if (member.isAdmin) {
+                        setFamilyManagerEmail(member.email);
+                        break;
+                    }
+                }
+            } else {
+                if (res.status === 401) {
+                    setAuthToken('');
+                    router.replace({ pathname: '/' });
+                } else {
+                    router.replace({
```

Comment:
so this will redirect him to index even on some network error

Reply:
can you check if we get a 400 or 500 range  error and then only redirect him to index

Reply:
or even simply not redirecting also will work

Reply:
just redirecting on invalid token should be fine